spring-integration-core/src/main/java/org/springframework/integration/config/annotation/AbstractMethodAnnotationPostProcessor.java

```diff
@@ -150,7 +150,7 @@ public Object postProcess(Object bean, String beanName, Method method, List<Anno
 					((Advised) handler).addAdvisor(advisor);
 				}
 				else {
-					ProxyFactory proxyFactory = new ProxyFactory(bean);
+					ProxyFactory proxyFactory = new ProxyFactory(handler);
```

Comment:
BTW, how have you catched it? Do you any specific test? 
I mean save the time for the backporting :smile: